voxblox/include/voxblox/skeletons/skeleton.h

```diff
@@ -0,0 +1,135 @@
+#ifndef VOXBLOX_SKELETONS_SKELETON_H_
+#define VOXBLOX_SKELETONS_SKELETON_H_
+
+#include <map>
+
+#include "voxblox/core/common.h"
+
+namespace voxblox {
+
+struct SkeletonPoint {
+  Point point = Point::Zero();
+  float distance = 0.0f;
+  int num_basis_points = 0;
+  AlignedVector<Point> basis_directions;
+
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+};
+
+class Skeleton {
+ public:
+  Skeleton();
+
+  // Access to all the skeleton points.
+  const AlignedVector<SkeletonPoint>& getSkeletonPoints() const {
+    return points_;
+  }
+  const AlignedVector<SkeletonPoint>& getEdgePoints() const { return edges_; }
+  const AlignedVector<SkeletonPoint>& getVertexPoints() const {
+    return vertices_;
+  }
+
+  AlignedVector<SkeletonPoint>& getSkeletonPoints() { return points_; }
```

Comment:
If you are offering full access to all the private variables, why not just make this a struct with these members public?